### PR TITLE
MXSession: Make `directRooms` atomic and copying

### PR DESCRIPTION
**Pull request by ismailgulek**
Fix vector-im/element-ios#4911

### CODE REVIEW

File: MatrixSDK/MXSession.m [hunk 1]
```diff
@@ -2635,9 +2637,9 @@ - (MXHTTPOperation*)uploadDirectRoomsInOperationsQueue:(NSDictionary<NSString *,
                                                failure:(void (^)(NSError *))failure
 {
     // If there is no change, do nothing
-    if (_directRooms == directRooms
-        || [_directRooms isEqualToDictionary:directRooms]
-        || (_directRooms == nil && directRooms.count == 0))
+    if (self.directRooms == directRooms
```
Comment (manuroe) [inline MatrixSDK/MXSession.m:2640]: Same remark for a temporary variable to host `self.directRooms`

File: MatrixSDK/MXSession.m [hunk 2]
```diff
@@ -1718,14 +1720,14 @@ - (void)handleAccountData:(NSDictionary*)accountDataUpdate
                 NSDictionary<NSString*, NSArray<NSString*>*> *directRooms;
                 MXJSONModelSetDictionary(directRooms, event[@"content"]);
 
-                if (directRooms != _directRooms
-                    && ![directRooms isEqualToDictionary:_directRooms])
+                if (directRooms != self.directRooms
```
Comment (manuroe) [inline MatrixSDK/MXSession.m:1723]: Can we use a temporary variable to host `self.directRooms` to limit atomic synchronisation that can lock the current thread?

**Review by manuroe — COMMENTED** [review]
Just a comment for probably a small perf improvement.

File: MatrixSDK/MXSession.m (hunk 1 above)
Comment (ismailgulek) [reply MatrixSDK/MXSession.m]: 👍

File: MatrixSDK/MXSession.m (hunk 2 above)
Comment (ismailgulek) [reply MatrixSDK/MXSession.m]: 👍

**Review by manuroe — APPROVED** [review]

**Pull request merged**